diff --git a/aim/web/ui/src/pages/Dashboard/Dashboard.tsx b/aim/web/ui/src/pages/Dashboard/Dashboard.tsx
index 1dbda66..6899d40 100644
--- a/aim/web/ui/src/pages/Dashboard/Dashboard.tsx
+++ b/aim/web/ui/src/pages/Dashboard/Dashboard.tsx
@@ -6,7 +6,6 @@ import { Spinner, Text } from 'components/kit';
 
 import ProjectContributions from './components/ProjectContributions/ProjectContributions';
 import ExploreSection from './components/ExploreSection/ExploreSection';
-import DashboardRight from './components/DashboardRight/DashboardRight';
 import DashboardContributionsFeed from './components/DashboardContributionsFeed';
 import ProjectStatistics from './components/ProjectStatistics';
 import useProjectContributions from './components/ProjectContributions/useProjectContributions';
@@ -54,7 +53,6 @@ function Dashboard(): React.FunctionComponentElement<React.ReactNode> {
           )}
           {!isLoading && !totalRunsCount && <AimIntegrations />}
         </div>
-        <DashboardRight />
       </section>
     </ErrorBoundary>
   );
diff --git a/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatistics.tsx b/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatistics.tsx
index b345cdc..35ef65b 100644
--- a/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatistics.tsx
+++ b/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatistics.tsx
@@ -22,61 +22,6 @@ const statisticsInitialMap: Record<string, IProjectStatistic> = {
     iconBgColor: '#7A4CE0',
     navLink: routes.METRICS.path,
   },
-  systemMetrics: {
-    label: 'Sys. metrics',
-    count: 0,
-    icon: 'metrics',
-    iconBgColor: '#AF4EAB',
-    navLink: `${routes.METRICS.path}?select=${encode({
-      advancedQuery: "metric.name.startswith('__system__') == True",
-      advancedMode: true,
-    })}`,
-  },
-  [SequenceTypesEnum.Figures]: {
-    label: 'Figures',
-    icon: 'figures',
-    count: 0,
-    iconBgColor: '#18AB6D',
-    navLink: routes.FIGURES_EXPLORER.path,
-  },
-  [SequenceTypesEnum.Images]: {
-    label: 'Images',
-    icon: 'images',
-    count: 0,
-    iconBgColor: '#F17922',
-    navLink: routes.IMAGE_EXPLORE.path,
-  },
-  [SequenceTypesEnum.Audios]: {
-    label: 'Audios',
-    icon: 'audios',
-    count: 0,
-    iconBgColor: '#FCB500',
-    navLink: routes.AUDIOS_EXPLORER.path,
-    badge: {
-      value: 'New',
-      style: { backgroundColor: '#1473e6', color: '#fff' },
-    },
-  },
-  [SequenceTypesEnum.Texts]: {
-    label: 'Texts',
-    icon: 'text',
-    count: 0,
-    iconBgColor: '#E149A0',
-    navLink: '',
-    badge: {
-      value: 'Explorer coming soon',
-    },
-  },
-  [SequenceTypesEnum.Distributions]: {
-    label: 'Distributions',
-    icon: 'distributions',
-    count: 0,
-    iconBgColor: '#0394B4',
-    navLink: '',
-    badge: {
-      value: 'Explorer coming soon',
-    },
-  },
 };
 
 const runsCountingInitialMap: Record<'archived' | 'runs', IProjectStatistic> = {
diff --git a/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatisticsStore.ts b/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatisticsStore.ts
index a9cca3a..effffcc 100644
--- a/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatisticsStore.ts
+++ b/aim/web/ui/src/pages/Dashboard/components/ProjectStatistics/ProjectStatisticsStore.ts
@@ -6,14 +6,7 @@ import { SequenceTypesEnum } from 'types/core/enums';
 function projectStatisticsEngine() {
   const { fetchData, state, destroy } = createResource<GetParamsResult>(() =>
     getParams({
-      sequence: [
-        SequenceTypesEnum.Metric,
-        SequenceTypesEnum.Images,
-        SequenceTypesEnum.Figures,
-        SequenceTypesEnum.Texts,
-        SequenceTypesEnum.Audios,
-        SequenceTypesEnum.Distributions,
-      ],
+      sequence: [SequenceTypesEnum.Metric],
       exclude_params: true,
     }),
   );
diff --git a/aim/web/ui/src/pages/Dashboard/components/QuickStart/QuickStart.tsx b/aim/web/ui/src/pages/Dashboard/components/QuickStart/QuickStart.tsx
index 4608789..cec61aa 100644
--- a/aim/web/ui/src/pages/Dashboard/components/QuickStart/QuickStart.tsx
+++ b/aim/web/ui/src/pages/Dashboard/components/QuickStart/QuickStart.tsx
@@ -10,6 +10,9 @@ import { DOCUMENTATIONS } from 'config/references';
 import './QuickStart.scss';
 
 function QuickStart() {
+  // Get fasttrack server hostname from current browser location
+  const { hostname, port } = window.location;
+  const fasttrack_server = `http://${hostname}:${port}`;
   return (
     <div className='QuickStart'>
       <Text
@@ -34,7 +37,7 @@ function QuickStart() {
           code={`from aim import Run
 
 # Initialize a new run
-run = Run()
+run = Run(repo=${fasttrack_server})
 
 # Log run parameters
 run["hparams"] = {
